src/libstd/keyword_docs.rs

```diff
@@ -1353,11 +1353,180 @@ mod super_keyword {}
 
 #[doc(keyword = "trait")]
 //
-/// A common interface for a class of types.
+/// A common interface for a group of types.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// A `trait` is an interface that types can implement. It is said they
```

Comment:
```suggestion
/// A `trait` is like an interface that data types can implement.
/// When a type implements a trait it can be treated abstractly 
/// as that trait using generics or trait objects.
```